### PR TITLE
feature/user-config

### PR DESCRIPTION
**Pull request by CenLX**
#### Code:
- [ ] Commits have descriptive messages
- [ ] Branch is rebased off of latest master
- [ ] Tests

---
### PR Review approvals:
- [x] Approval 1
- [x] Approval 2

### CODE REVIEW

File: config/config.global.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var config = module.exports = {}
```
Comment (Duelist) [inline config/config.global.js]: Move `module.exports` to the bottom to stay consistent.

```
module.exports = config
```

File: config/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+var env = process.env.NODE_ENV || 'development'
+try
```
Comment (Duelist) [inline config/index.js]: `try {`

File: config/index.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+{
+  var cfg = require('./config.' + env)
+}
+catch (e)
```
Comment (Duelist) [inline config/index.js]: `catch (e) {`

File: config/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+var env = process.env.NODE_ENV || 'development'
+try
+{
+  var cfg = require('./config.' + env)
```
Comment (Duelist) [inline config/index.js]: Use full var names similar to your other files. `config`

File: config/config.global.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+
+config.env = 'development'
+
+config.discordjs = {}
```
Comment (Duelist) [inline config/config.global.js]: The token itself isn't specific to discord.js, so `discord`. Might be switching away from discord.js I think.

File: config/index.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+}
+catch (e)
+{
+  if (e.code !== 'MODULE_NOT_FOUND') {
```
Comment (Duelist) [inline config/index.js]: Save an op here by flipping the conditions. `=== 'MODULE_NOT_FOUND'`

**Comment by Duelist** [comment]
Few minor changes, otherwise lgtm.

File: config/config.development.js.sample [hunk 7]
```diff
@@ -0,0 +1,5 @@
+var config = require('./config.global')
+
+config.discord.key = '<Insert Key Here>'
```
Comment (Fikaf) [inline config/config.development.js.sample]: maybe call it discord.channel_key or something , .key doesnt explain what kind of key.

Comment (CenLX) [inline config/config.development.js.sample]: It's the bot's login key.

Comment (Fikaf) [inline config/config.development.js.sample]: maybe server_token or just token

**Comment by Fikaf** [comment]
╱╱┏╮
╱╱┃┃
▉━╯┗━╮
▉┈┈┈┈┃
▉╮┈┈┈┃
╱╰━━━╯

**Pull request merged**